qiskit/qpy/formats.py

```diff
@@ -85,6 +85,26 @@
 CIRCUIT_INSTRUCTION_PACK = "!HHHII?Hq"
 CIRCUIT_INSTRUCTION_SIZE = struct.calcsize(CIRCUIT_INSTRUCTION_PACK)
 
+# CIRCUIT_INSTRUCTION_V2
+CIRCUIT_INSTRUCTION_V2 = namedtuple(
+    "CIRCUIT_INSTRUCTION",
+    [
+        "name_size",
+        "label_size",
+        "num_parameters",
+        "num_qargs",
+        "num_cargs",
+        "has_condition",
+        "condition_register_size",
+        "condition_value",
+        "num_ctrl_qubits",
+        "ctrl_state",
```

Comment:
Why this is an integer rather than string or byte?

Reply:
I used an integer because that's what the internal state field of the object uses. It accepts a bitstring as a user input but converts that to an int in the setter. I guess you're right it would be better to use an arbitrary length bit string here since this will error if someone tries to set a control state on >64 qubits (python won't care but qpy will). Not sure how likely that is in practice at least in the near term, but if you want I can change the type here and do the conversion.